provider/openstack/init.go

```diff
@@ -19,4 +20,14 @@ func init() {
 	tools.RegisterToolsDataSourceFunc("keystone catalog", getKeystoneToolsSource)
 
 	registry.RegisterEnvironStorageProviders(providerType)
+
+	// Register the Openstack specific providers.
+	registry.RegisterProvider(
+		provider.CinderProviderType,
+		&provider.OpenstackProvider{provider.NewGooseAdapter},
+	)
+
+	// Inform the storage provider registry about the Openstack provider.
```

Comment:
Comment is a little weird. It's recording in the registry which storage providers are supported by OpenStack.